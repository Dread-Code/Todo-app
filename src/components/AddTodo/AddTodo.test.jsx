import React from "react";
import { shallow } from "enzyme";
import { AddTodo } from "./AddTodo";

describe("AddTodo", () => {
  const addTodo = jest.fn();
  const wrapper = shallow(<AddTodo addTodo={addTodo} />);

  test("should  match snapshoot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("shouldn't call handleAddTodo", () => {
    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });
    expect(addTodo).toHaveBeenCalledTimes(0);
  });

  test("should call handleAddTodo", () => {
    const value = "Estudiar Enzyme";

    wrapper.find("input").simulate("change", {
      target: { name: "desc", value },
    });

    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });

    expect(addTodo).toHaveBeenCalledTimes(1);
    expect(addTodo).toHaveBeenCalledWith(expect.any(Object));
    expect(addTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      desc: value,
      done: false,
    });
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
