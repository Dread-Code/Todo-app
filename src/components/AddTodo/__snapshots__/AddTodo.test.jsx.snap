// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo should  match snapshoot 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="desc"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
